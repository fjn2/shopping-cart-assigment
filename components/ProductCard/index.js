import Card from "react-bootstrap/Card"

const ProductCard = ({ id, name, brand, price, available }) => {
  return (
    <a href={`details/${id}`} style={{ color: 'inherit', textDecoration: 'none'}}>
      <Card style={{ display: 'flex'}}>
        <Card.Img style={{ filter: `grayscale(${available ? 0 : 1})` }} variant="top" src={`https://avatars.dicebear.com/api/jdenticon/${name}_${price}.svg`} />
        <Card.Body>
          <Card.Title>{available ? '' : '[X] '}{name}</Card.Title>
          <Card.Text>
            <b>{brand}</b>
          </Card.Text>
          <span>{price} sek</span>
        </Card.Body>
      </Card>
    </a>
  )
}

export default ProductCard
