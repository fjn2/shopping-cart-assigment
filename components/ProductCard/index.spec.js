import { shallow } from 'enzyme';
import ProductCard from './';

describe('ProductCard', () => {
  it('Should render correctly', () => {
    const props = {
      id: 10,
      name: 'Test product',
      brand: 'Test brand name',
      price: 124,
      available: true
    }

    const elem = shallow(<ProductCard {...props} />)
    
    expect(elem).toMatchSnapshot()
  })
  it('Should render an unavailable product', () => {
    const props = {
      id: 10,
      name: 'Test product',
      brand: 'Test brand name',
      price: 124,
      available: false
    }

    const elem = shallow(<ProductCard {...props} />)
    
    expect(elem).toMatchSnapshot()
  })
})