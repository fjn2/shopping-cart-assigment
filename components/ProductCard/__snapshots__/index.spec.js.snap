// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Should render an unavailable product 1`] = `
<a
  href="details/10"
  style={
    Object {
      "color": "inherit",
      "textDecoration": "none",
    }
  }
>
  <Card
    body={false}
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <CardImg
      src="https://avatars.dicebear.com/api/jdenticon/Test product_124.svg"
      style={
        Object {
          "filter": "grayscale(1)",
        }
      }
      variant="top"
    />
    <CardBody>
      <CardTitle>
        [X] 
        Test product
      </CardTitle>
      <CardText>
        <b>
          Test brand name
        </b>
      </CardText>
      <span>
        124
         sek
      </span>
    </CardBody>
  </Card>
</a>
`;

exports[`ProductCard Should render correctly 1`] = `
<a
  href="details/10"
  style={
    Object {
      "color": "inherit",
      "textDecoration": "none",
    }
  }
>
  <Card
    body={false}
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <CardImg
      src="https://avatars.dicebear.com/api/jdenticon/Test product_124.svg"
      style={
        Object {
          "filter": "grayscale(0)",
        }
      }
      variant="top"
    />
    <CardBody>
      <CardTitle>
        Test product
      </CardTitle>
      <CardText>
        <b>
          Test brand name
        </b>
      </CardText>
      <span>
        124
         sek
      </span>
    </CardBody>
  </Card>
</a>
`;
